nixos/modules/services/mail/dovecot.nix

```diff
@@ -7,7 +7,90 @@ let
   dovecotPkg = pkgs.dovecot;
 
   baseDir = "/run/dovecot2";
-  stateDir = "/var/lib/dovecot";
+
+  sieveToPath = text:
+    if isString text && builtins.substring 0 2 text == "/*"
+    then pkgs.writeText "script.sieve" text
+    else if types.path.check text then text
+    else pkgs.writeText "script.sieve" text;
+  scriptType = types.coercedTo types.lines sieveToPath types.path;
+
+  compileSieveScripts = pkgs.buildSieveScripts {
+    name = "dovecot2-sieveScripts";
+      plugins = cfg.sieve.plugins;
+      extensions = cfg.sieve.extensions;
+      globalExtensions = cfg.sieve.globalExtensions;
+      phases = ["buildPhase" "installPhase" "fixupPhase"];
```

Comment:
It is recommended to use `dontConfigure` instead of defining your own list of phases.

Reply:
done! But needed to also set `dontUnpack`